standalone/include/mbi-tools.h

```diff
@@ -22,14 +22,33 @@
 #include <stddef.h>
 #include <stdbool.h>
 #include <mbi.h>
+#include <elf.h>
 
-bool mbi_find_memory(const struct mbi *multiboot_info, size_t len,
-                     void **block_start, size_t *block_len,
-                     bool highest);
+void exclude_region(uint64_t *block_addr, uint64_t *block_len,
+                    uintptr_t const image_start, uintptr_t const image_end);
+void exclude_bender_binary(uint64_t *block_addr, uint64_t *block_len);
+bool overlap_bender_binary(struct ph64 const * p);
 
-void *mbi_alloc_protected_memory(struct mbi *multiboot_info, size_t len, unsigned align);
+static inline bool overlap(uint64_t mod_start, uint64_t mod_end,
+                           struct ph64 const * p)
+{
+  if (p->p_memsz == 0)
+    return false;
+
+  uint64_t const p_first = p->p_paddr;
+  uint64_t const p_last  = p->p_paddr + p->p_memsz - 1;
+
+  return !((mod_end < p_first) || (p_last < mod_start));
```

Comment:
The function is not used consistently. In at least one invocation mod_end is one byte after the last valid byte, in others it's the last valid byte. In the first case, the `overlap()` function would have an off-by-one error. I would propose to unify this to the half-open interval style: [begin, begin + size). Then the correct function would look like this:

```C
static inline bool overlap(uint64_t mod_start, uint64_t mod_end,
                           struct ph64 const * p)
{
  if (p->p_memsz == 0 || mod_end <= mod_start)
    return false;

  uint64_t const p_first = p->p_paddr;
  uint64_t const p_end   = p->p_paddr + p->p_memsz;

  return !((mod_end <= p_first) || (p_end <= mod_start));
}
```